docs/readmes/nms/dev_testing.md

```diff
@@ -0,0 +1,29 @@
+---
+id: dev_testing
+title: Testing Tips
+hide_title: true
+---
+
+# Testing Tips
+
+This guide covers tips for quickly validating NMS changes.
+
+## Run Test on the host
+
+There are some simple tests that you should run after making changes.
+
+### Eslint
+Run `cd ${MAGMA_ROOT}/nms/app/ && yarn run eslint` to test that there are no linter errors.
+You can also run `cd ${MAGMA_ROOT}/nms/app/ && yarn run eslint --fix` to fix automatically
+fixable linter errors.
+
+### Flow
+Run `cd ${MAGMA_ROOT}/nms/app/ && yarn run flow` to test that there are no type errors.
+
+### Unit Test
+Run `cd ${MAGMA_ROOT}/nms/app/ && yarn run test` to test that your changes have not broken existing
+invididual units / components. This command also runs any new unit tests that you created.
+
+### End to End Test
+This is not a mandatory test. You can run `cd ${MAGMA_ROOT}/nms/app/ && yarn run test:e2e` to test that your
```

Comment:
Do you know why this isn't mandatory?

Reply:
iirc they were a bit flaky

Reply:
It is not mandatory since it is not required in github codereview process.
<img width="810" alt="Screen Shot 2021-07-21 at 10 50 42 AM" src="https://user-images.githubusercontent.com/34602743/126517097-1e3d8d0f-d0b3-4619-8e7a-d4166aa039c0.png">
But I have also added a precommit section now which makes it more clear. The other thing is that first time developer, who download magma should expect all the tests to pass if she/he didn't make any change. However, this e2e test will not pass because it expects other things: one thing I saw that is, it expects a list of pictures(assets). 
<img width="1507" alt="Screen Shot 2021-07-21 at 11 38 20 AM" src="https://user-images.githubusercontent.com/34602743/126517591-340d6408-8736-486e-a257-990f8e70d057.png">
 Tbh, whenever I run that test locally it fails saying timeout error, image missing or other error, but it still passes on CI.  So, I wanted to make it clear that you don't have run this test.

Reply:
Got it, that sounds great thanks Dani! Seems like we should fix the e2e test or provide a doc section on common flake failure patterns - can be a separate Issue/PR though